core-bundle/src/EventListener/DataContainer/TrackTitleSourceListener.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener\DataContainer;
+
+use Contao\CoreBundle\DependencyInjection\Attribute\AsCallback;
+use Contao\CoreBundle\Filesystem\FilesystemUtil;
+use Contao\CoreBundle\Filesystem\VirtualFilesystem;
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\DataContainer;
+use Contao\Message;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Session\FlashBagAwareSessionInterface;
+
+#[AsCallback(table: 'tl_content', target: 'fields.textTrackSRC.load')]
+#[AsCallback(table: 'tl_content', target: 'fields.textTrackSRC.save')]
+readonly class TrackTitleSourceListener
+{
+    public function __construct(
+        private ContaoFramework $framework,
+        private VirtualFilesystem $filesStorage,
+        private RequestStack $requestStack,
+    ) {
+    }
+
+    public function __invoke(mixed $value, DataContainer $dc): mixed
+    {
+        $invalid = [];
+
+        if (null !== $value) {
+            $fileSystemItems = FilesystemUtil::listContentsFromSerialized($this->filesStorage, $value);
+
+            foreach ($fileSystemItems as $fileSystemItem) {
+                $extraMetadata = $fileSystemItem->getExtraMetadata();
+
+                if (
+                    null === $extraMetadata->getTextTrack()?->getSourceLanguage()
+                    || '' === $extraMetadata->getLocalized()?->getFirst()?->getTitle()
+                ) {
+                    $invalid[] = $fileSystemItem->getName();
+                }
+            }
+        }
+
+        $message = $this->framework->getAdapter(Message::class);
+
+        if ([] !== $invalid) {
+            $message->addError(\sprintf($GLOBALS['TL_LANG']['ERR']['textTrackMetadataMissing'], implode(', ', $invalid)));
+        } elseif ($message->hasError()) {
+            $session = $this->requestStack->getCurrentRequest()?->getSession();
+
+            // Reset the error if it exists
+            if ($session instanceof FlashBagAwareSessionInterface) {
+                $session->getFlashBag()->get('contao.BE.error');
+            }
+        }
```

Comment:
Why is this needed? Only adding the message should be fine.

Reply:
It's for removing the previous message when adjusting the files.

Reply:
Hm, that should not be the case. Also this would apply to _any_ error, not just the one you added. Also why does simply calling `$session->getFlashBag()->get('contao.BE.error');` reset it?

Reply:
^ Basically because it sets an empty Array of messages to the flashbag so that is the intended way to use it.

Reply:
Yes, but you should just use `$message->reset()` then. In any case, I would inspect why the flashbag is displayed again, this should not happen. Also I would make the following additional changes:

1. Remove `submitOnChange` from `tl_content.textTrackSRC`.
2. Remove `#[AsCallback(table: 'tl_content', target: 'fields.textTrackSRC.save')]` from `TrackTitleSourceListener`.

Reply:
See 5af6ce9f2e5cd1365e2b2285fabcc418fb56ea92

Added a `ToDo` as mentioned in Slack due to the message being kept even within the next request